lib/collective.lib.js

```diff
@@ -232,6 +232,14 @@ export const isSelfHostedAccount = c => c.isHost === true && c.type === 'COLLECT
 export const isIndividualAccount = account => ['USER', 'INDIVIDUAL'].includes(account.type);
 
 /* Checks whether an account supports grants */
-export const accountSupportsGrants = (collectiveSettings, hostSettings) => {
-  return get(collectiveSettings, 'expenseTypes.hasGrant', !get(hostSettings, 'disableGrantsByDefault', false));
+export const accountSupportsGrants = (account, host) => {
+  if (get(account, 'settings.expenseTypes.hasGrant')) {
+    // If the opt-in flag in ON, always allow grants
+    return true;
+  } else if (get(host, 'settings.disableGrantsByDefault')) {
+    // If the host has configured grants to be opt-in, the previous flag (`hasGrant`) **must** be set
+    return false;
+  } else {
+    return [CollectiveType.FUND, CollectiveType.PROJECT].includes(account?.type);
+  }
```

Comment:
I think we need to also add the `!get(account, 'settings.expenseTypes.hasGrant')` condition as otherwise even when the user toggles the checkbox to turn off grants it would not take effect. 🤔 

```suggestion
  if (get(account, 'settings.expenseTypes.hasGrant') === true) {
    // If the opt-in flag in ON, always allow grants
    return true;
  } else if (get(host, 'settings.expenseTypes.hasGrant') === false) {
   // If the opt-in flag in OFF, do not allow grants
    return false;
  } else if (get(host, 'settings.disableGrantsByDefault')) {
    // If the host has configured grants to be opt-in, the previous flag (`hasGrant`) **must** be set
    return false;
  } else {
    return [CollectiveType.FUND, CollectiveType.PROJECT].includes(account?.type);
  }
```

Reply:
Good catch, pushing another fix for that

Reply:
No worries. Thanks a lot for helping me with this; I should have got this right in the last PR. 🥲